From 80ec2b01027c25236eee1fc0a7468a17d480e931 Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@nxp.com>
Date: Mon, 5 Mar 2018 05:27:05 -0600
Subject: [PATCH 3426/5242] MLK-17685-1 Enable CCI perfomance monitor

commit  35183fdffe37fcf5477a31bf21efb7ef52619733 from
https://source.codeaurora.org/external/imx/linux-imx.git

perf list

  CCI_400_r1/cycles/                                 [Kernel PMU event]
  CCI_400_r1/mi_retry_speculative_fetch,source=?/    [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_barrier_hazard,source=?/   [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_hi_prio_rtq_full,source=?/ [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_low_prio_rtq_full,source=?/ [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_master_id_hazard,source=?/ [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_mid_prio_rtq_full,source=?/ [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_qvn_vn0,source=?/          [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_qvn_vn1,source=?/          [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_qvn_vn2,source=?/          [Kernel PMU event]
  CCI_400_r1/mi_rrq_stall_qvn_vn3,source=?/          [Kernel PMU event]

Signed-off-by: Frank Li <Frank.Li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi |   21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 5a60c68..82ae55d1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -148,6 +148,27 @@
 			     <0 32 4>, <0 32 4>, <0 32 4>, <0 32 4>;
 	};
 
+	cci@52090000 {
+		compatible = "arm,cci-400";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0 0x52090000 0 0x1000>;
+		ranges = <0 0 0x52090000 0x10000>;
+
+		pmu@9000 {
+			compatible = "arm,cci-400-pmu,r1",
+				     "arm,cci-400-pmu";
+			reg = <0x9000 0x4000>;
+			interrupts = <GIC_SPI 8 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 10 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 11 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 12 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 13 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-parent = <&gic>;
+		};
+	};
+
 	imx8qm-pm {
 		#address-cells = <1>;
 		#size-cells = <0>;
-- 
1.7.9.5

